### ISSUE TITLE
Im@s event farming

### ISSUE THREAD

**Issue by cobeng13** [Masuzu/ZooeyBot#93]
I've been trying to setup the bot to farm the imas events, Particularly the 20AP Raid event battle. I enabled event mode and put the support summon page as the url. 

However once I start the bot, after calibration it does nothing or it crashes, unlike the previous event that the bot will paste the url at the address bar. Am I doing something wrong?

http://imgur.com/VrcihId - screenshot

**Comment by ariaanaa** [comment]
i have the same problem here, for event it always crash but if i set the parameter to false for farm other things it just work normally

**Comment by MagCat** [comment]
@ariaanaa: can you elaborate your solution little bit more? I set the Event Mode parameter to false and with normal mode, the bot still does nothing or crash when selects Event Quest.

**Comment by ariaanaa** [comment]
@MagCat well i didn't test it for the current event, i just test it with angel halo, and other quest in the favorite tab that just work fine for me

**Comment by MagCat** [comment]
@ariaanaa oh okay, I misread your sentences then. I thought you figured a workaround solution for the event. It's a shame that Masuzu had already announced before he won't put more major support until late June so chance is that we might not even see the solution to this event.

**Comment by Dreamfinal** [comment]
me too, has stop crashes

**Comment by ariaanaa** [comment]
@MagCat do you have the older version? maybe it can work with that. 
I just tried event parameter to set "true" and it work and didn't crash, I use ver. 1.6.3 that the latest i have so far

**Comment by Dreamfinal** [comment]
[EventMode]
# Set this parameter to true to use the event mode. Zooey will enter autopilot mode when you navigate to the quest menu.
Enabled=true
# Url of the summon selection page for Angel Halo (Extreme)
EventRaidUrl=http://game.granbluefantasy.jp/#quest/supporter/510031/5
EventRaidScript=Scripts/test.lua
# Url of the summon selection page for Dimensional Halo
NightmareModeUrl=http://game.granbluefantasy.jp/#quest/supporter/510051/5
NightmareModeScript=
# Some events such as Rise of the Beasts require to navigate to the home page of the event before navigating to the event raid url defined above. If EventPageUrl is provided, Zooey will first navigate to this url before hosting the raid. Leave EventPageUrl blank for events which do not require you to do so.
EventPageUrl=
# Comma-delimited list of summons which will be selected in priority if they are available for the nightmare mode.
# In case none of the preferred summons was found, select a random summon in the summon attribute tab defined by the parameter DefaultSummonAttributeTab, unless the parameter RerollSummonWhenNoPreferredSummonWasFoundForNightmareMode is set to true.
NightmareModePreferredSummons=Lucifer
# Zooey will reroll summons until one of the preferred summons for the nightmare mode is available, using trial battles to refresh the summon list.
RerollSummonWhenNoPreferredSummonWasFoundForNightmareMode=true
# Set this parameter to true if the nightmare mode is available when you start Zooey so she can proceed to the nightmare stage right ahead.
NightmareModeAvailableAtStart=false
# Wait time in milliseconds after Zooey finishes loading the EventPageUrl
WaitTimeInMsAfterEventPageIsLoaded=1000

 @ariaanaa 

Do you have a link for download 1.6.3?
Can I try?

I use ver 1.9.3.
Can you try my version to see if it has any errors?

**Comment by Dreamfinal** [comment]
- `if battle < 3 then
-   DisableChargeAttack()
- else
-   EnableChargeAttack()
- end
- if battle == 1 and turn == 1 then
-   Summon(6)
- end
- if battle == 1 and turn == 1 then
-   characters["Heles"]:UseSkill(3)
-   characters["Jeanne d'Arc"]:UseSkill(3)
- end
- if battle == 3 then
-   characters["Gran"]:UseSkill(2)
-   characters["Gran"]:UseSkill(4)
-   characters["Jeanne d'Arc"]:UseSkill(1)
-   characters["Jeanne d'Arc"]:UseSkill(2)
-   characters["Heles"]:UseSkill(1)
-   characters["Heles"]:UseSkill(2)
-   characters["Lucio"]:UseSkill(1)
-   characters["Lucio"]:UseSkill(3)
- end`

**Comment by okamijackie** [comment]
what is a basic lua script to use skills for fights i am not sure how to set it up here are my characters S.zooey, vira, narumaya

**Comment by Dreamfinal** [comment]
Mode Slime Slave = Normal
Mode TreasureEventMode = Normal
Mode EventMode = has stop

**Comment by cobeng13** [comment]
Tried ver 1.5.3 it works

**Comment by MagCat** [comment]
Can anyone upload version 1.5.3-1.6.3? We really appreciate if you also write down what kind of input you put in parameter to make it work as well.

**Comment by ariaanaa** [comment]
here for [ver. 1.6.3](https://mega.nz/#!IsQCSDCD!jwu2wXUw5983hqgW7N5biRoS8zUYcIKPWzWWTJnySDU) for input parameter just set it to true, and change the link for current event like imas, here is my setting http://i.imgur.com/o86emZs.png

**Comment by Dreamfinal** [comment]
@ariaanaa 

Hey gay 1.6.3 it works. Thank you

the same lua skip version 1.9.3.

**Comment by MagCat** [comment]
@ariaanaa I'm having trouble to get my script working. The bot just keeps using basic attacks instead of executing the script to use skills.

```
DisableChargeAttack()

if battle == 1 and turn == 1 then
character_4.UseSkill(2)
character_3.UseSkill(2)
character_3.UseSkill(1)
character_2:UseSkill(2)
end

if battle == 2 and turn == 2 then
Summon(6)
end

if battle == 3 and turn == 1 then
character_2.UseSkill(4)
end

if battle == 3 and turn >= 2 and character_2.charge_gauge >= 90 and character_3.charge_gauge >= 80 and character_4.charge_gauge >= 70 then
EnableChargeAttack()
character_1:UseSkill(2)
character_1:UseSkill(1)
character_1:UseSkill(3)
character_2:UseSkill(1)
character_3:UseSkill(3)
end
```
This is what I set up for Event parameters, I tried your parameters and for some reasons the bot can't get either script working (client crash) or getting stuck in selecting right summons. 
```
[EventMode]
# Set this parameter to true to use the event mode. Zooey will enter autopilot mode when you navigate to the quest menu.
Enabled=true
# Url of the summon selection page for Angel Halo (Extreme)
EventRaidUrl=http://game.granbluefantasy.jp/#quest/supporter/719281/5
EventRaidScript=Scripts/Eventidol_BerserkerPT.lua
# Url of the summon selection page for Dimensional Halo
NightmareModeUrl=http://game.granbluefantasy.jp/#quest/supporter/719281/5
NightmareModeScript=Scripts/Eventidol_BerserkerPT.lua
# Some events such as Rise of the Beasts require to navigate to the home page of the event before navigating to the event raid url defined above. If EventPageUrl is provided, Zooey will first navigate to this url before hosting the raid. Leave EventPageUrl blank for events which do not require you to do so.
EventPageUrl=
# Comma-delimited list of summons which will be selected in priority if they are available for the nightmare mode.
# In case none of the preferred summons was found, select a random summon in the summon attribute tab defined by the parameter DefaultSummonAttributeTab, unless the parameter RerollSummonWhenNoPreferredSummonWasFoundForNightmareMode is set to true.
NightmareModePreferredSummons=Bahamut
# Zooey will reroll summons until one of the preferred summons for the nightmare mode is available, using trial battles to refresh the summon list.
RerollSummonWhenNoPreferredSummonWasFoundForNightmareMode=true
# Set this parameter to true if the nightmare mode is available when you start Zooey so she can proceed to the nightmare stage right ahead.
NightmareModeAvailableAtStart=false
# Wait time in milliseconds after Zooey finishes loading the EventPageUrl
WaitTimeInMsAfterEventPageIsLoaded=1000
```

**Comment by meteoshower** [comment]
same here 1.9.3 was crash.i back to use 1.8.1 and work fine

**Comment by i-Grey** [comment]
@meteoshower Could you upload 1.8.1 too? I would like to try it

**Comment by Masuzu** [comment]
@MagCat The syntax should be `character_4:UseSkill(2)`

I encountered some crash with the bot at the start of the turn if you ran into that same issue in the past. I will fix this in v1.9.4.
If `NightmareModeScript` is not provided the bot will use the default behaviour (basic attacks). Same if the `kEventRaidScript` parameter is left blank.

**Comment by Masuzu** [comment]
https://poker-bot.azurewebsites.net/ZooeyBot/en/Home/PatchNotes
v1.9.7 is now compatible again with events.
I apologize for the inconvenience.
